run.py

```diff
@@ -74,16 +85,49 @@ def model_state(country, state, interventions=None):
     }
     return CovidTimeseriesModel().forecast(model_parameters=MODEL_PARAMETERS)
 
-if __name__ == '__main__':
-    logging.basicConfig(level=logging.INFO)
-    Dataset = CDSDataset()
-    for state in Dataset.get_all_states_by_country('USA'):
-        for i in range(0, len(INTERVENTIONS)):
-            intervention = INTERVENTIONS[i]
+
+@click.group()
+def main():
+    """Entrypoint for Data Model CLI."""
+    pass
+
+
+@main.command()
+@click.argument('dataset_name', type=click.Choice(['JHU', 'CDS']), default="CDS")
+@click.option(
+    '--output-dir', '-o', default='results/test', help="Model results output directory"
+)
+@click.option('--country', default='USA')
+@click.option('--state', '-s', multiple=True)
+def run_model(dataset_name, output_dir, country, state):
```

Comment:
I guess `@click` is doing some magic to automatically run this function if you pass `run_model` in on the command-line?  That's neat, but is non-obvious (to somebody like me at least).  I have a slight preference against using libraries that do too much magic until you need to.

Though if we rework the CLI interface to remove "run-model" from the command-line, as I suggested above, maybe this will go away?

In any case, I'm kind of on the fence about this...